parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -1150,8 +1150,11 @@ public void writeDataPageV2(
     int rlByteLength = toIntWithCheck(repetitionLevels.size(), "page repetition levels");
     int dlByteLength = toIntWithCheck(definitionLevels.size(), "page definition levels");
 
-    int compressedSize =
-        toIntWithCheck(compressedData.size() + repetitionLevels.size() + definitionLevels.size(), "page");
+    int compressedSize = 0;
+    if (compressedData.size() > 0) {
+      compressedSize =
+          toIntWithCheck(compressedData.size() + repetitionLevels.size() + definitionLevels.size(), "page");
+    }
```

Comment:
So this class is duplicating code from `ColumnChunkPageWriteStore.java` above? Do you know why that is?

Reply:
`ParquetFileWriter` supports writing out pages directly. So there are some duplicate codes. I plan to reduce the duplication for better maintenance.